src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,144 @@
+
+import java.time.LocalDateTime;
+
+@Tag(name = "RecommendationRequest")
+@RequestMapping("/api/recommendationrequest")
+@RestController
+@Slf4j
+public class RecommendationRequestController extends ApiController {
+    @Autowired
+    RecommendationRequestRepository recommendationRequestRepository;
+    @Autowired
+    CurrentUserService currentUserService;
+
+    /**
+     * List all recommendation requests
+     * 
+     * @return an iterable of RecommendationRequest
+     */
+    @Operation(summary= "List all recommendation requests")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @GetMapping("/alladmin")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> requests = recommendationRequestRepository.findAll();
+        return requests;
+    }
+	
+	/**
+     * List all recommendation requests created by a user with requesterId
+     * 
+     * @return an iterable of RecommendationRequest
+     */
+    @Operation(summary= "List all recommendation requests")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> getAllCurrentUser() {
+        User currentUser = currentUserService.getUser();
+        Long requesterId = currentUser.getId();
+        Iterable<RecommendationRequest> requests = recommendationRequestRepository.findAllByRequesterId(requesterId);
+
+        return requests;
+    }
+
+    /**
+     * Create a new request
+     * 
+     * @param requesterId   
+     * @param professorEmail     
+     * @param requestType      
+     * @param details          
+     * @param neededByDate 
+     * @param submissionDate
+     * @param completionDate
+     * @param status
+     * @return                  
+     */
+    @Operation(summary= "Create a new request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping("/post")
+
+    public RecommendationRequest postRecommendationRequest(
+            @Parameter(name="professorId") @RequestParam Long professorId,
+            @Parameter(name="requestType") @RequestParam String requestType,
+            @Parameter(name="details") @RequestParam String details,
+            @Parameter(name="neededByDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("neededByDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime neededByDate
+            )
+            throws JsonProcessingException {
+
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+        // See: https://www.baeldung.com/spring-date-parameters
+
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        User currentUser = currentUserService.getUser();
+        Long requesterId = currentUser.getId();
+
+        recommendationRequest.setRequesterId(requesterId);
+        recommendationRequest.setProfessorId(professorId);
```

Comment:
Since this is coming from user input, it needs to be validated, i.e. check that this user id is in the users table, and that the user it refers to has the "professor" role.